### PR TITLE
CASMTRIAGE-3326 simplify clock skew check by using a chronyc command

### PR DESCRIPTION
**Pull request by jacobsalmela**
Signed-off-by: Jacob Salmela <jacob.salmela@hpe.com>

## Summary and Scope

Dramatically simplifies the clock skew test by running a chronyc command.

This command waits until the node is in sync.  In this test, I check every 10 seconds, timing out after a minute and allowing for skew of 50ms (50,000ppm).

This does away with all the SSH stuff and just talks directly to chrony to see if it thinks it is in sync.    The various healthcheck scripts target the nodes, so when we have wrapper upon wrapper, the ssh does a lot of overhead.

## Issues and Related PRs

* Resolves CASMTRIAGE-3326'

### Tested on:

  * `#hela`
  *  `#redbull`
  *  `#mug`

### Test description:

Ran manually:

```
ncn-m001:~ # chronyc waitsync 6 0.1 0.1 10
try: 1, refid: 0AFC010D, correction: 0.000001236, skew: 0.006
ncn-m001:~ # echo $?
0
ncn-m001:~ #
```

Ran locally through GOSS:

```
ncn-m001:~ #  GOSS_BASE=/opt/cray/tests/install/ncn/ goss -g ./c.yml --vars /opt/cray/tests/install/ncn/vars/variables-ncn.yaml validate --max-concurrent 4
.

Total Duration: 0.007s
Count: 1, Failed: 0, Skipped: 0
```

Ran through ncn-healthchecks:

```
ncn-m001:~ # /opt/cray/tests/install/ncn/automated/ncn-healthcheck
NCN Run-Time Checks
-------------------

Running tests against node ncn-s001
Server URL: http://ncn-s001.hmn:9001/ncn-healthcheck-storage
...
...
Result: PASS
Test Name: Check clock skew on k8s and storage nodes
Description: If this test fails, see "operations/node_management/Configure_NTP_on_NCNs.md#fix-broken-configs" in the CSM release documentation for steps to remediate clock skew.
Severity: 0
Test Summary: Command: check_clock_skew: exit-status: matches expectation: [0]
Execution Time: 2.02431234 seconds
Node: ncn-m003
...
...
```

## Risks and Mitigations

Low.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by heemstra — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

File: goss-testing/tests/ncn/goss-check-clock-skew.yaml [hunk 1]
```diff
@@ -22,19 +22,18 @@
 # OTHER DEALINGS IN THE SOFTWARE.
 #
 
-{{ $scripts := .Env.GOSS_BASE | printf "%s/scripts" }}
-{{ $logrun := $scripts | printf "%s/log_run.sh" }}
-{{ $check_clock_skew := $scripts | printf "%s/check_clock_skew.sh" }}
 command:
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-check-clock-skew.yaml:25]:
```suggestion
{{ $scripts := .Env.GOSS_BASE | printf "%s/scripts" }}
{{ $logrun := $scripts | printf "%s/log_run.sh" }}
command:
```

File: goss-testing/tests/ncn/goss-check-clock-skew.yaml [hunk 2]
```diff
@@ -22,19 +22,18 @@
...
+    title: Check clock skew on k8s and storage nodes
+    meta:
+      desc: If this test fails, see "operations/node_management/Configure_NTP_on_NCNs.md#fix-broken-configs" in the CSM release documentation for steps to remediate clock skew.
+      sev: 0
+    # check to see if chrony is in sync using a chronyc command
+    # up to 6 times
+    # with a max correction of 0.5ppm (50ms)
+    # and a max skew of 0.5ppm (50ms)
+    # retry every 10 seconds
+    exec: "chronyc waitsync 6 0.5 0.5 10"
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-check-clock-skew.yaml:36]:
```suggestion
        exec: |-
            "{{$logrun}}" -l "{{$testlabel}}" \
                /usr/bin/chronyc waitsync 6 0.5 0.5 10
```

File: goss-testing/tests/ncn/goss-check-clock-skew.yaml [hunk 3]
```diff
@@ -22,19 +22,18 @@
...
-        meta:
-            desc: If this test fails, see "operations/node_management/Configure_NTP_on_NCNs.md#fix-broken-configs" in the CSM release documentation for steps to remediate clock skew.
-            sev: 0
-        exec: |-
-            "{{$logrun}}" -l "{{$testlabel}}" \
-                "{{$check_clock_skew}}"
-        exit-status: 0
-        timeout: 60000
-        skip: false
+  check_clock_skew:
```
Comment (mharding-hpe) [inline goss-testing/tests/ncn/goss-check-clock-skew.yaml:26]:
```suggestion
    {{ $testlabel := "check_clock_skew" }}
    {{$testlabel}}:
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
Is there a reason we wouldn't want to log the output of the command? It seems like that could be useful, particularly in the case of a failure.

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
Should this PR also delete the current clock skew shell script, since no tests are using it any longer?

File: goss-testing/tests/ncn/goss-check-clock-skew.yaml (hunk 1 above)
Comment (jacobsalmela) [reply goss-testing/tests/ncn/goss-check-clock-skew.yaml:26]: Fixed, scripts dir isn't needed

File: goss-testing/tests/ncn/goss-check-clock-skew.yaml (hunk 2 above)
Comment (jacobsalmela) [reply goss-testing/tests/ncn/goss-check-clock-skew.yaml]: Fixed

File: goss-testing/tests/ncn/goss-check-clock-skew.yaml (hunk 3 above)
Comment (jacobsalmela) [reply goss-testing/tests/ncn/goss-check-clock-skew.yaml]: fixed

**Comment by mharding-hpe** [comment]
Also need a main branch PR for this.

**Comment by jacobsalmela** [comment]
@mharding-hpe https://github.com/Cray-HPE/csm-testing/pull/315

**Pull request merged**